Building on Windows fails with Lua installed

The build process fails both for 1.5.0 and for the current master branch in an MSYS2 environment on Windows if Lua is installed in MSYS2.

Steps to reproduce:

1. Install Java JDK on Windows (may not be necessary)
2. Install MSYS2 and launch with `msys2_shell.cmd -mingw64 -full-path`
3. Install packages:
```
pacman -S \
mingw64/mingw-w64-x86_64-toolchain \
mingw64/mingw-w64-x86_64-cmake \
mingw64/mingw-w64-x86_64-glib2 \
glib2-devel \
mingw64/mingw-w64-x86_64-gtest \
mingw64/mingw-w64-x86_64-lua \
mingw64/mingw-w64-x86_64-doxygen \
mingw64/mingw-w64-x86_64-python-sphinx
```
4. From the LCM directory, run `cmake -B build -DPython_FIND_REGISTRY=NEVER`
5. Build: `cmake --build build`

Output of step 4:
```
-- Building for: Ninja
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/msys64/mingw64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found GLib2_glib: C:/msys64/mingw64/lib/libglib-2.0.dll.a
-- Found GLib2: C:/msys64/mingw64/lib/libglib-2.0.dll.a  found components: glib
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__std_gnu99
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__std_gnu99 - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wall
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wall - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wshadow
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wshadow - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wuninitialized
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wuninitialized - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wunused_variable
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wunused_variable - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Werror_return_type
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Werror_return_type - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wno_unused_parameter
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wno_unused_parameter - Success
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wno_format_zero_length
-- Performing Test CMAKE_C_COMPILER_SUPPORTS__Wno_format_zero_length - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wall
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wall - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wshadow
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wshadow - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wuninitialized
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wuninitialized - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wunused_variable
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wunused_variable - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Werror_return_type
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Werror_return_type - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wno_unused_parameter
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wno_unused_parameter - Success
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wno_format_zero_length
-- Performing Test CMAKE_CXX_COMPILER_SUPPORTS__Wno_format_zero_length - Success
-- Building LCM 1.5.0 (ABI v1)
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Found Python: C:/msys64/mingw64/bin/python3.12.exe (found version "3.12.7") found components: Interpreter
-- Found Python: C:/msys64/mingw64/bin/python3.12.exe (found version "3.12.7") found components: Interpreter Development Development.Module Development.Embed
-- Found Java: C:/Program Files/Java/jdk-23/bin/java.exe (found suitable version "23.0.1", minimum required is "1.8")
-- Found Lua: C:/msys64/mingw64/lib/liblua.dll.a (found version "5.4.7")
-- Found Go: C:/Program Files/Go/bin/go.exe (found version "1.23.4")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Deprecation Warning at lcm-cmake/lcmUtilities.cmake:32 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
Call Stack (most recent call first):
  test/types/CMakeLists.txt:1 (include)


-- CPack: Packages will be placed under C:/Users/jg110/Downloads/lcm-1.5.0/build/packages
-- Configuring done (23.5s)
-- Generating done (0.2s)
-- Build files have been written to: C:/Users/jg110/Downloads/lcm-1.5.0/build
```

Output of step 5:
```
[21/136] Building Java objects for jchart2d.jar
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
[48/136] Building Java objects for lcm-java.jar
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
[126/136] Linking C shared module lcm-lua\lcm.dll
FAILED: lcm-lua/lcm.dll
C:\Windows\system32\cmd.exe /C "cd . && C:\msys64\mingw64\bin\cc.exe -std=gnu99 -Wno-format-zero-length   -shared -o lcm-lua\lcm.dll -Wl,--major-image-version,0,--minor-image-version,0 lcm-lua/CMakeFiles/lcm-lua.dir/init.c.ob
j lcm-lua/CMakeFiles/lcm-lua.dir/lualcm_hash.c.obj lcm-lua/CMakeFiles/lcm-lua.dir/lualcm_lcm.c.obj lcm-lua/CMakeFiles/lcm-lua.dir/lualcm_pack.c.obj lcm-lua/CMakeFiles/lcm-lua.dir/lua_ref_helper.c.obj lcm-lua/CMakeFiles/lcm-lu
a.dir/utf8_check.c.obj  lcm/liblcm.dll.a  C:/msys64/mingw64/lib/liblua.dll.a  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: lcm-lua/CMakeFiles/lcm-lua.dir/lualcm_lcm.c.obj:lualcm_lcm.c:(.text+0x751): undefined reference to `__imp_select'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: lcm-lua/CMakeFiles/lcm-lua.dir/lualcm_lcm.c.obj:lualcm_lcm.c:(.text+0x78f): undefined reference to `__WSAFDIsSet'
collect2.exe: error: ld returned 1 exit status
[131/136] Linking CXX executable test\c\test-c-memq_test.exe
ninja: build stopped: subcommand failed.
```

I'm able to get a successful build if I disable the Lua components by replacing the command for step 4 with `cmake -B build -DLCM_ENABLE_LUA=OFF -DPython_FIND_REGISTRY=NEVER`

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -166,7 +166,8 @@ jobs:
           cmake:p
           glib2:p
           gtest:p
-  
+          lua:p
+
     - name: Configure CMake
       run: cmake -B build -DPython_FIND_REGISTRY=NEVER
 

--- a/lcm-lua/CMakeLists.txt
+++ b/lcm-lua/CMakeLists.txt
@@ -16,6 +16,10 @@ target_include_directories(lcm-lua PRIVATE
 )
 
 target_link_libraries(lcm-lua lcm ${LUA_LIBRARY})
+if (WIN32)
+  target_link_libraries(lcm-lua ws2_32)
+endif()
+
 
 install(TARGETS lcm-lua
   DESTINATION lib${LIB_SUFFIX}/lua/${LUA_VERSION_MAJOR}.${LUA_VERSION_MINOR}

